### PR TITLE
Guard against NullPointerException when getting Credentials from Json

### PR DESCRIPTION
**Pull request by bennycao**
### Changes

This change adds code to guard against possible NullPointerException thrown as a result of `f` being null in the JsonRequiredTypeAdaptor

Solves https://github.com/auth0/Auth0.Android/issues/676 

### Checklist

- [ ] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)

- [ ] I have read the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md)

- [ ] All existing and new tests complete without errors

### CODE REVIEW

**Comment by poovamraj** [comment]
@bennycao I'd rather prefer us to check if the field is null [here](https://github.com/auth0/Auth0.Android/blob/c17b87b4f67ddec78e5e6be9276c6f84a7bcd402/auth0/src/main/java/com/auth0/android/request/internal/JsonRequiredTypeAdapterFactory.java#L35) like below

```
if (f!= null && f.getAnnotation(JsonRequired.class) != null) {
```

Instead of catching the exception. It feels much cleaner and the check makes more sense. Can you try this solution out?

**Comment by bennycao** [comment]
> @bennycao yes if the required value like `expiresAt` is not present for constructing the object, there will be an error thrown, but since this is fixing an issue that no one is able to reproduce, this is the only change that will introduce no new changes to the public API but at the same time solve your issue. If making this change introduces the new issue in your error logs, atleast you will be able to further debug it.

@poovamraj i've made the update as requested.

**Comment by bennycao** [comment]
@poovamraj Any chance we can get this looked thanks ?

**Review by poovamraj — APPROVED** [review]

**Comment by bennycao** [comment]
@poovamraj thanks for the approval. When do we think a new release with this fix be available ?
In terms of build status, i'm not sure what is happening there. But let me know if i need to do anything. Thanks

**Comment by bennycao** [comment]
@poovamraj any movement on getting this into a release ? we would like to close this out asap. Thanks

**Comment by poovamraj** [comment]
@bennycao you can go ahead with merging the PR after ensuring all checks are passed. We can cut a patch release once this is merged.

**Comment by bennycao** [comment]
> @bennycao you can go ahead with merging the PR after ensuring all checks are passed. We can cut a patch release once this is merged.

@poovamraj it seems some build steps require authorizations to run. And i don't have rights to merge. Can you please authorize these ?

File: auth0/src/main/java/com/auth0/android/request/internal/JsonRequiredTypeAdapterFactory.java [hunk 1]
```diff
@@ -32,7 +32,7 @@ public T read(JsonReader in) throws IOException {
 
                 Field[] fields = pojo.getClass().getDeclaredFields();
                 for (Field f : fields) {
-                    if (f.getAnnotation(JsonRequired.class) != null) {
```
Comment (RMSPORTS1902) [inline auth0/src/main/java/com/auth0/android/request/internal/JsonRequiredTypeAdapterFactory.java:35]:
```suggestion
                    if (f.getAnnotation(JsonRequired.class) != null) {
```

Comment (RMSPORTS1902) [reply auth0/src/main/java/com/auth0/android/request/internal/JsonRequiredTypeAdapterFactory.java:35]: Yes

**Pull request merged**